promptfoo/promptfoo
# Add OpenAI factuality and closed-QA graders

Add the factuality and closed QA model-graded evaluation prompts from OpenAI's eval toolkit.  They work similarly to `llm-rubric`, but presumably are more battle-tested by OpenAI.

Two new assertion types:
- `model-graded-factuality`
- `model-graded-closedqa`

Example usage:

```yaml
prompts: [name_capitals.txt]
tests:
  - vars:
      location: Sacramento

    assert:
      # Ensure the answer agrees with the provided facts
      - type: model-graded-factuality
        value: The capital of California is Sacramento

      # Ensure the answer meets the criteria
      - type: model-graded-closedqa
        value: Does not include any extra information or facts about Sacramento
```
